code/computational_geometry/src/sphere_tetrahedron_intersection/sphere_tetrahedron_intersection.cpp

```diff
@@ -0,0 +1,1237 @@
+	}
+
+	// VECTOR *= VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T >& operator *=( LuVector< N, T >& lhs, const LuVector< N, S >& rhs )
+	{
+		UNROLL( lhs, rhs, OP_MUL< T, S >(), LOOP_INDEX< N - 1 >() );
+		return lhs;
+	}
+
+	// VECTOR /= VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T >& operator /=( LuVector< N, T >& lhs, const LuVector< N, S >& rhs )
+	{
+		UNROLL( lhs, rhs, OP_DIV< T, S >(), LOOP_INDEX< N - 1 >() );
+		return lhs;
+	}
+
+	//////////////////////////////////////////////////////////////
+
+	// VECTOR + SCALAR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator +( const LuVector< N, T >& lhs, const S& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result += rhs;
+	}
+
+	// SCALAR + VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator +( const T& lhs, const LuVector< N, S >& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result += rhs;
+	}
+
+	// VECTOR - SCALAR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator -( const LuVector< N, T >& lhs, const S& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result -= rhs;
+	}
+
+	// SCALAR - VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator -( const T& lhs, const LuVector< N, S >& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result -= rhs;
+	}
+
+	// VECTOR * SCALAR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator *( const LuVector< N, T >& lhs, const S& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result *= rhs;
+	}
+
+	// SCALAR * VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator *( const T& lhs, const LuVector< N, S >& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result *= rhs;
+	}
+
+	// VECTOR / SCALAR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator /( const LuVector< N, T >& lhs, const S& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result /= rhs;
+	}
+
+	// SCALAR / VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator /( const T& lhs, const LuVector< N, S >& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result /= rhs;
+	}
+
+	//////////////////////////////////////////////////////////////
+
+	// VECTOR + VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator +( const LuVector< N, T >& lhs, const LuVector< N, S >& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result += rhs;
+	}
+
+	// VECTOR - VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator -( const LuVector< N, T >& lhs, const LuVector< N, S >& rhs )
+	{
+		LuVector< N, T > result( lhs );
+		return result -= rhs;
+	}
+
+	// VECTOR * VECTOR
+	template< size_t N, class T, class S >
+	inline LuVector< N, T > operator *( const LuVector< N, T >& lhs, const LuVector< N, S >& rhs )
+	{
+		LuVector< N, T > result( lhs );
```

Comment:
We use 4 spaces, not tabs.

Reply:
Seriously? That's absolutely barbaric!
jk :)
Why though?

Reply:
By the way, what do you say about this: https://github.com/OpenGenus/cosmos/issues/3541

Reply:
@RedBlight Since we don't have the option of a separate directory for just includes, an ugly option may have to be considered: Directly including .cpp files using relative paths. Yes, I know, I know, you're probably screaming at the mere thought of doing this, but there really isn't any more choice. Most of the algorithms in this repo were meant to be simple 1 file things, but now that we are expanding, we should make the appropriate changes. For now, there isn't much you can do.

Reply:
"...Directly including .cpp files using relative paths..."
Actually, seperating the vector class into a seperate header file is not that bad.
I'll just add another file as "LuVector.hpp" containing the vector class, and then include it in "sphere_tetrahedron_intersection.cpp" if that's OK?

In that case, can we treat LuVector as a black box and leave it's formatting as it is? Since it will just be used as a header-only library in the main code file.